extensions/cl_exp_tensor.asciidoc

```diff
@@ -0,0 +1,1058 @@
+  command and can be used to query or queue a wait for this command to
+  complete. If _event_ is NULL or the enqueue is unsuccessful, no
+  event will be created and therefore it will not be possible to query
+  the status of this command or to wait for this command to
+  complete. If _event_wait_list_ and _event_ are not NULL, _event_
+  must not refer to an element of the _event_wait_list_ array.
+
+Elements are copied from the source tensor to the destination tensor
+so that after the completion following condition holds expressed in
+pseudo C:
+
+[source,c]
+----
+// 'so' and 'do' are aliases for src_origin and dst_origin, respectively.
+tensor_element(dst_tensor, do[0] + i[0], do[1] + i[1], ..., do[N-1] + i[N-1])
+==
+tensor_element(src_tensor, so[0] + i[0], so[1] + i[1], ..., so[N-1] + i[N-1]);
+----
+
+Where the `N` is tensor rank, the `i[X]` is a tensor coordinate with
+inclusive range of `0..<region[X]-1>`.
+
+*clEnqueueCopyTensorEXP* returns CL_SUCCESS if the function is
+executed successfully. Otherwise, it returns one of the following
+errors:
+
+* CL_INVALID_COMMAND_QUEUE if _command_queue_ is not a valid host
+  command-queue.
+
+* CL_INVALID_CONTEXT if the context associated with _command_queue_
+  and buffer are not the same or if the context associated with
+  _command_queue_ and events in _event_wait_list_ are not the same.
+
+* CL_INVALID_MEM_OBJECT if _src_tensor_ or _dst_tensor_ are not a
+  valid buffer object created with `CL_MEM_TENSOR_EXP`.
+
+* CL_INVALID_VALUE if _tensor_origin_ or _mem_origin_ is NULL.
+
+* CL_INVALID_VALUE if _src_origin_, _dst_origin_ or _region_ is NULL.
+
+* CL_INVALID_VALUE if `region[i]` is zero for i in `[0, tensor_rank)`.
+
+* CL_INVALID_VALUE if `origin[i] + region[i] > tensor_shape[i]` at any
+  dimension `i` in range `[0, tensor_rank)`.
+
+* CL_INVALID_EVENT_WAIT_LIST if _event_wait_list_ is NULL and
+  _num_events_in_wait_list_ > 0, or _event_wait_list_ is not NULL and
+  _num_events_in_wait_list_ is 0, or if event objects in
+  _event_wait_list_ are not valid events.
+
+* CL_EXEC_STATUS_ERROR_FOR_EVENTS_IN_WAIT_LIST if the read and write
+  operations are blocking and the execution status of any of the
+  events in _event_wait_list_ is a negative integer value.
+
+* CL_MEM_OBJECT_ALLOCATION_FAILURE if there is a failure to allocate
+  memory for data store associated with memory object the _tensor_ is
+  bound to.
+
+* CL_OUT_OF_RESOURCES if there is a failure to allocate resources
+  required by the OpenCL implementation on the device.
+
+* CL_OUT_OF_HOST_MEMORY if there is a failure to allocate resources
+  required by the OpenCL implementation on the host.
+
+// TODO: add clEnqueueFillTensor?
+
+--
+// End (Add the following to Section 5.2.2, *Reading, Writing and Copying Buffer Objects*)
+
+
+(Add the following to Section 5.17.5, *Recording Commands to a Command-Buffer*) ::
++
+--
+
+If *cl_khr_command_buffer* is supported, then the following command
```

Comment:
Refer to this extension in the intro spec, state that optionally cl_khr_command_buffer is extended.